File: app/src/main/java/vgame/ir/data/remote/model/CoursesResponse.java
package vgame.ir.data.remote.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class CoursesResponse {
    @SerializedName("error")
    @Expose
    private Boolean error;
    @SerializedName("message")
    @Expose
    private String message;
    @SerializedName("data")
    @Expose
    private Data data;

    public Boolean getError() {
        return error;
    }

    public void setError(Boolean error) {
        this.error = error;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public Data getData() {
        return data;
    }

    public void setData(Data data) {
        this.data = data;
    }

    public class Data {

        @SerializedName("courses")
        @Expose
        private CourseResponse courseResponse;

        @SerializedName("chapters")
        @Expose
        private ChapterResponse chapterResponse;

        @SerializedName("sessions")
        @Expose
        private LessonResponse lessonResponse;

        public CourseResponse getCourseResponse() {
            return courseResponse;
        }

        public void setCourseResponse(CourseResponse courseResponse) {
            this.courseResponse = courseResponse;
        }

        public ChapterResponse getChapterResponse() {
            return chapterResponse;
        }

        public void setChapterResponse(ChapterResponse chapterResponse) {
            this.chapterResponse = chapterResponse;
        }

        public LessonResponse getLessonResponse() {
            return lessonResponse;
        }

        public void setLessonResponse(LessonResponse lessonResponse) {
            this.lessonResponse = lessonResponse;
        }
    }

    public class CourseResponse {

        @SerializedName("data")
        @Expose
        private List<Course> coursesList;

        public List<Course> getCoursesList() {
            return coursesList;
        }

        public void setCoursesList(List<Course> coursesList) {
            this.coursesList = coursesList;
        }
    }

    public class ChapterResponse {

        @SerializedName("data")
        @Expose
        private List<CourseDetailResponse.Chapter> chapterList;
    }

    public class LessonResponse {

        @SerializedName("data")
        @Expose
        private List<LessonDetailResponse.Session> sessionList;

        public List<LessonDetailResponse.Session> getSessionList() {
            return sessionList;
        }

        public void setSessionList(List<LessonDetailResponse.Session> sessionList) {
            this.sessionList = sessionList;
        }
    }

    public class Courses {

        @SerializedName("current_page")
        @Expose
        private Integer currentPage;
        @SerializedName("data")
        @Expose
        private List<Course> data = null;
        @SerializedName("first_page_url")
        @Expose
        private String firstPageUrl;
        @SerializedName("from")
        @Expose
        private Integer from;
        @SerializedName("last_page")
        @Expose
        private Integer lastPage;
        @SerializedName("last_page_url")
        @Expose
        private String lastPageUrl;
        @SerializedName("next_page_url")
        @Expose
        private String nextPageUrl;
        @SerializedName("path")
        @Expose
        private String path;
        @SerializedName("per_page")
        @Expose
        private Integer perPage;
        @SerializedName("prev_page_url")
        @Expose
        private Object prevPageUrl;
        @SerializedName("to")
        @Expose
        private Integer to;
        @SerializedName("total")
        @Expose
        private Integer total;

        public Integer getCurrentPage() {
            return currentPage;
        }

        public void setCurrentPage(Integer currentPage) {
            this.currentPage = currentPage;
        }

        public List<Course> getData() {
            return data;
        }

        public void setData(List<Course> data) {
            this.data = data;
        }

        public String getFirstPageUrl() {
            return firstPageUrl;
        }

        public void setFirstPageUrl(String firstPageUrl) {
            this.firstPageUrl = firstPageUrl;
        }

        public Integer getFrom() {
            return from;
        }

        public void setFrom(Integer from) {
            this.from = from;
        }

        public Integer getLastPage() {
            return lastPage;
        }

        public void setLastPage(Integer lastPage) {
            this.lastPage = lastPage;
        }

        public String getLastPageUrl() {
            return lastPageUrl;
        }

        public void setLastPageUrl(String lastPageUrl) {
            this.lastPageUrl = lastPageUrl;
        }

        public String getNextPageUrl() {
            return nextPageUrl;
        }

        public void setNextPageUrl(String nextPageUrl) {
            this.nextPageUrl = nextPageUrl;
        }

        public String getPath() {
            return path;
        }

        public void setPath(String path) {
            this.path = path;
        }

        public Integer getPerPage() {
            return perPage;
        }

        public void setPerPage(Integer perPage) {
            this.perPage = perPage;
        }

        public Object getPrevPageUrl() {
            return prevPageUrl;
        }

        public void setPrevPageUrl(Object prevPageUrl) {
            this.prevPageUrl = prevPageUrl;
        }

        public Integer getTo() {
            return to;
        }

        public void setTo(Integer to) {
            this.to = to;
        }

        public Integer getTotal() {
            return total;
        }

        public void setTotal(Integer total) {
            this.total = total;
        }

    }

    public class Teacher {

        @SerializedName("id")
        @Expose
        private Integer id;
        @SerializedName("description")
        @Expose
        private String description;
        @SerializedName("user_id")
        @Expose
        private Integer userId;
        @SerializedName("full_name")
        @Expose
        private String fullName;
        @SerializedName("avatar_url")
        @Expose
        private String avatarUrl;

        private String hight_light;


        public String getHight_light() {
            return hight_light;
        }

        public void setHight_light(String hight_light) {
            this.hight_light = hight_light;
        }

        public Integer getId() {
            return id;
        }

        public void setId(Integer id) {
            this.id = id;
        }

        public String getDescription() {
            return description;
        }

        public void setDescription(String description) {
            this.description = description;
        }

        public Integer getUserId() {
            return userId;
        }

        public void setUserId(Integer userId) {
            this.userId = userId;
        }

        public String getFullName() {
            return fullName;
        }

        public void setFullName(String fullName) {
            this.fullName = fullName;
        }

        public String getAvatarUrl() {
            return avatarUrl;
        }

        public void setAvatarUrl(String avatarUrl) {
            this.avatarUrl = avatarUrl;
        }

    }

    public class Category {

        @SerializedName("id")
        @Expose
        private Integer id;
        @SerializedName("name")
        @Expose
        private String name;
        @SerializedName("major_id")
        @Expose
        private Integer majorId;
        @SerializedName("created_at")
        @Expose
        private String createdAt;
        @SerializedName("updated_at")
        @Expose
        private String updatedAt;
        @SerializedName("is_active")
        @Expose
        private Integer isActive;

        public Integer getId() {
            return id;
        }

        public void setId(Integer id) {
            this.id = id;
        }

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }

        public Integer getMajorId() {
            return majorId;
        }

        public void setMajorId(Integer majorId) {
            this.majorId = majorId;
        }

        public String getCreatedAt() {
            return createdAt;
        }

        public void setCreatedAt(String createdAt) {
            this.createdAt = createdAt;
        }

        public String getUpdatedAt() {
            return updatedAt;
        }

        public void setUpdatedAt(String updatedAt) {
            this.updatedAt = updatedAt;
        }

        public Integer getIsActive() {
            return isActive;
        }

        public void setIsActive(Integer isActive) {
            this.isActive = isActive;
        }

    }
}

File: app/src/main/java/vgame/ir/data/remote/model/CoursesResponse.kt
package vgame.ir.data.remote.model

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class CoursesResponse {
    @SerializedName("error")
    @Expose
    var error: Boolean? = null
    @SerializedName("message")
    @Expose
    var message: String? = null
    @SerializedName("data")
    @Expose
    var data: Data? = null

    inner class Data {

        @SerializedName("courses")
        @Expose
        var courseResponse: CourseResponse? = null

        @SerializedName("chapters")
        @Expose
        var chapterResponse: ChapterResponse? = null

        @SerializedName("sessions")
        @Expose
        var lessonResponse: LessonResponse? = null
    }

    inner class CourseResponse {

        @SerializedName("data")
        @Expose
        var coursesList: List<Course>? = null
    }

    inner class ChapterResponse {

        @SerializedName("data")
        @Expose
        private val chapterList: List<CourseDetailResponse.Chapter>? = null
    }

    inner class LessonResponse {

        @SerializedName("data")
        @Expose
        var sessionList: List<LessonDetailResponse.Session>? = null
    }

    inner class Courses {

        @SerializedName("current_page")
        @Expose
        var currentPage: Int? = null
        @SerializedName("data")
        @Expose
        var data: List<Course>? = null
        @SerializedName("first_page_url")
        @Expose
        var firstPageUrl: String? = null
        @SerializedName("from")
        @Expose
        var from: Int? = null
        @SerializedName("last_page")
        @Expose
        var lastPage: Int? = null
        @SerializedName("last_page_url")
        @Expose
        var lastPageUrl: String? = null
        @SerializedName("next_page_url")
        @Expose
        var nextPageUrl: String? = null
        @SerializedName("path")
        @Expose
        var path: String? = null
        @SerializedName("per_page")
        @Expose
        var perPage: Int? = null
        @SerializedName("prev_page_url")
        @Expose
        var prevPageUrl: Any? = null
        @SerializedName("to")
        @Expose
        var to: Int? = null
        @SerializedName("total")
        @Expose
        var total: Int? = null

    }

    inner class Teacher {

        @SerializedName("id")
        @Expose
        var id: Int? = null
        @SerializedName("description")
        @Expose
        var description: String? = null
        @SerializedName("user_id")
        @Expose
        var userId: Int? = null
        @SerializedName("full_name")
        @Expose
        var fullName: String? = null
        @SerializedName("avatar_url")
        @Expose
        var avatarUrl: String? = null

        var hight_light: String? = null

    }

    inner class Category {

        @SerializedName("id")
        @Expose
        var id: Int? = null
        @SerializedName("name")
        @Expose
        var name: String? = null
        @SerializedName("major_id")
        @Expose
        var majorId: Int? = null
        @SerializedName("created_at")
        @Expose
        var createdAt: String? = null
        @SerializedName("updated_at")
        @Expose
        var updatedAt: String? = null
        @SerializedName("is_active")
        @Expose
        var isActive: Int? = null

    }
}
